torch/functional.py

```diff
@@ -1062,6 +1063,11 @@ def norm(input, p="fro", dim=None, keepdim=False, out=None, dtype=None):  # noqa
         >>> torch.norm(d[0, :, :]), torch.norm(d[1, :, :])
         (tensor(3.7417), tensor(11.2250))
     """
+    warn(("Deprecation warning: The interface and functionality of `norm` will "
+          "be changed in the future, to be consistent with `numpy.linalg.norm`."
+          " This change is planned for PyTorch 1.7. Check our issue tracker for"
+          " details: https://github.com/pytorch/pytorch/issues/24802"),
+         stacklevel=2)
```

Comment:
I don't think an unconditional warning is the right thing to do here. We should only warn on the conditions which would actually trigger behavior change in the future.

Another good test: there should be a way to fix the warning when you see it. How do I fix the warning with just this PR?